src/main/java/org/elasticsearch/search/highlight/HighlightPhase.java

```diff
@@ -46,11 +45,19 @@
 public class HighlightPhase extends AbstractComponent implements FetchSubPhase {
 
     private final Highlighters highlighters;
+    private final ImmutableList<String> sortedHighlighters;
 
     @Inject
     public HighlightPhase(Settings settings, Highlighters highlighters) {
         super(settings);
         this.highlighters = highlighters;
+        // TODO: would be nice if each highlighter would have a precedence and we could just sort here by this value
+        // we could then also add custom highlighters to this list and they could be selected automatically depending on precendence
```

Comment:
can you remove this TODO? I'm not sure we are going to implement this after all, nobody needs it for now :)